### PR TITLE
Do not store modal contents in component's state

### PR DESCRIPTION
**Pull request by hazolsky**

### CODE REVIEW

**Comment by hazolsky** [comment]
While doing this seems to be a good idea
https://github.com/LiveTyping/android-colors/blob/05fcbbaeeb7074e2b1b6e3de007ec70b1c541f63/src/components/CopyCodeModal/CopyCodeModal.js#L22-L27
in fact it prevents updating generated android theme when you change colors in palette.

To solve this, we either need to update `state.currentText` each time component receives new props, or not keep that in store at all. This PR contains changes that implement that approach.

File: src/components/CopyCodeModal/CopyCodeModal.js [hunk 1]
```diff
@@ -36,35 +29,36 @@ class Modal extends Component {
...
       return (
         <div
-          key={name}
+          key={tabName}
           tabIndex={0}
           role="button"
-          className={`modal__tab ${selectedClass}`}
-          onClick={() => this.onTabChange(content, name)}
+          className={classNames(className)}
+          onClick={() => this.onTabChange(tabName)}
```
Comment (n-filatov) [inline src/components/CopyCodeModal/CopyCodeModal.js]: Recently read that use arrow functions in render method is bad practise, coz every execution of `render` method creating new instance of this function and its bad for performance. And if u want to pass data in calling function u must create component like `Tab` pass this function in props and call it with data( tabName ) from this component. [Link](https://daveceddia.com/avoid-bind-when-passing-props/) on the article.

**Review by n-filatov — CHANGES_REQUESTED** [review]

**Comment by hazolsky** [comment]
@Muffassa good catch 👍 
Does it look better now?

**Comment by hazolsky** [comment]
Hmm, i was wrong in the implementation above - it won't work. Give me more time to think.

**Comment by hazolsky** [comment]
There you go @Muffassa

**Review by n-filatov — APPROVED** [review]

**Pull request merged**